bitnami/mariadb/10.9/debian-11/rootfs/opt/bitnami/scripts/libmariadb.sh

```diff
@@ -265,20 +265,20 @@ mysql_initialize() {
     if is_file_writable "$DB_CONF_FILE"; then
         info "Updating 'my.cnf' with custom configuration"
         mysql_update_custom_config
-    else
-        warn "The ${DB_FLAVOR} configuration file '${DB_CONF_FILE}' is not writable. Configurations based on environment variables will not be applied for this file."
-    fi
-
-    if [[ -f "${DB_CONF_DIR}/my_custom.cnf" ]]; then
-        if is_file_writable "${DB_CONF_DIR}/bitnami/my_custom.cnf"; then
-            info "Injecting custom configuration 'my_custom.cnf'"
-            cat "${DB_CONF_DIR}/my_custom.cnf" > "${DB_CONF_DIR}/bitnami/my_custom.cnf"
-            if ! grep --silent "!include ${DB_CONF_DIR}/bitnami/my_custom.cnf" "${DB_CONF_FILE}"; then
-                echo "!include ${DB_CONF_DIR}/bitnami/my_custom.cnf" >> "${DB_CONF_FILE}"
+        # Including my_custom.cnf also requires write access to my.cnf
+        if [[ -f "${DB_CONF_DIR}/my_custom.cnf" ]]; then
+            if is_file_writable "${DB_CONF_DIR}/bitnami/my_custom.cnf"; then
+                info "Injecting custom configuration 'my_custom.cnf'"
+                cat "${DB_CONF_DIR}/my_custom.cnf" > "${DB_CONF_DIR}/bitnami/my_custom.cnf"
+                if ! grep --silent "!include ${DB_CONF_DIR}/bitnami/my_custom.cnf" "${DB_CONF_FILE}"; then
+                    echo "!include ${DB_CONF_DIR}/bitnami/my_custom.cnf" >> "${DB_CONF_FILE}"
+                fi
+            else
+                warn "Could not inject custom configuration for the ${DB_FLAVOR} configuration file '$DB_CONF_DIR/bitnami/my_custom.cnf' because it is not writable."
             fi
-        else
-            warn "Could not inject custom configuration for the ${DB_FLAVOR} configuration file '$DB_CONF_DIR/bitnami/my_custom.cnf' because it is not writable."
         fi
+    else
+        warn "The ${DB_FLAVOR} configuration file '${DB_CONF_FILE}' is not writable. Configurations based on environment variables will not be applied for this file."
     fi
```

Comment:
Hi @fivaldi,

Nice catch with the $DB_CONF_FILE write permissions!

Nevertheless, it is not true that write permissions are required for the entire code-block, as there could be the case where a read-only config file already contains the my_custom.cnf include sentence (`!include ${DB_CONF_DIR}/bitnami/my_custom.cnf`) and the entire code block would be ignored.

I would like to suggest a different approach here:

```suggestion
    if is_file_writable "$DB_CONF_FILE"; then
        info "Updating 'my.cnf' with custom configuration"
        mysql_update_custom_config
    else
        warn "The ${DB_FLAVOR} configuration file '${DB_CONF_FILE}' is not writable. Configurations based on environment variables will not be applied for this file."
    fi

    if [[ -f "${DB_CONF_DIR}/my_custom.cnf" ]]; then
        if is_file_writable "${DB_CONF_DIR}/bitnami/my_custom.cnf"; then
            info "Injecting custom configuration 'my_custom.cnf'"
            cat "${DB_CONF_DIR}/my_custom.cnf" > "${DB_CONF_DIR}/bitnami/my_custom.cnf"
            if ! grep --silent "!include ${DB_CONF_DIR}/bitnami/my_custom.cnf" "${DB_CONF_FILE}"; then
                if is_file_writable "$DB_CONF_FILE"; then
                    echo "!include ${DB_CONF_DIR}/bitnami/my_custom.cnf" >> "${DB_CONF_FILE}"
                else
                    warn "Could not inject custom configuration. The ${DB_FLAVOR} configuration file '${DB_CONF_FILE}' is not writable."
                fi
            fi
        else
            warn "Could not inject custom configuration for the ${DB_FLAVOR} configuration file '$DB_CONF_DIR/bitnami/my_custom.cnf' because it is not writable."
        fi
    fi
```

This applies to all branches of mariabd, not only '10.9'

Reply:
This suggestion is not applicable in GitHub. :(

Can you please send me the patch via some pastebin? 

![image](https://user-images.githubusercontent.com/23337584/216374515-2f80c0f0-699a-490c-a6b5-25fea3924990.png)